apps/studio/components/interfaces/Settings/Vault/Secrets/AddNewSecretModal.tsx

```diff
@@ -11,18 +15,25 @@ interface AddNewSecretModalProps {
 }
 
 const AddNewSecretModal = ({ visible, onClose }: AddNewSecretModalProps) => {
-  const { vault, ui } = useStore()
+  const { ui } = useStore()
   const [showSecretValue, setShowSecretValue] = useState(false)
   const [selectedKeyId, setSelectedKeyId] = useState<string>()
+  const { project } = useProjectContext()
 
-  const keys = vault.listKeys()
+  const { mutateAsync: addKeyMutation } = usePgSodiumKeyCreateMutation()
+  const { mutateAsync: addSecret } = useVaultSecretCreateMutation()
+
+  const { data: keys } = usePgSodiumKeysQuery({
+    projectRef: project?.ref!,
```

Comment:
For queries, `projectRef` should be nullable and then handled by the `enabled` flag.